### PR TITLE
fix(api): remove openid client

### PR DESCRIPTION
**Pull request by timoklingenhoefer**

### CODE REVIEW

File: packages/api/package.json [hunk 1]
```diff
@@ -32,15 +32,15 @@
...
     "eventsource": "^2.0.2",
     "form-data": "^4.0.0",
+    "jose": "^4.8.3",
     "jwt-decode": "^3.1.2",
-    "openid-client": "^5.1.8",
     "p-queue": "^6.6.2",
     "ts-mixer": "^6.0.1"
   },
   "devDependencies": {
+    "@mswjs/interceptors": "^0.17.3",
```
Comment (3fd) [inline packages/api/package.json:41]: @mswjs/interceptors is not used?

File: packages/api/lib/token-set.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  constructor(values) {
+    Object.assign(this, values);
+  }
+
+  set expires_in(value) {
+    this.expiresAt = Date.now() + Number(value);
+  }
+
+  get expires_in() {
+    return Math.max.apply(null, [this.expiresAt - Date.now(), 0]);
```
Comment (3fd) [inline packages/api/lib/token-set.ts:14]: why are you using an array here?
why not `Math.max(this.expiresAt - Date.now(), 0)` ?

File: packages/api/lib/http.service.ts [hunk 3]
```diff
@@ -87,17 +89,89 @@ export class HttpClient {
...
+      !issuer.grant_types_supported?.includes('client_credentials') ||
+      !issuer.token_endpoint_auth_methods_supported?.includes('client_secret_jwt')
+    ) {
+      throw new Error('Issuer does not support client_secret_jwt');
+    }
+
+    return issuer;
+  }
+
+  public async discoverIssuer(uri: string) {
```
Comment (3fd) [inline packages/api/lib/http.service.ts:110]:
```suggestion
  private async discoverIssuer(uri: string): Promise<Issuer> {
```

File: packages/api/lib/http.service.ts [hunk 4]
```diff
@@ -87,17 +89,89 @@ export class HttpClient {
...
+    const wellKnownUri = `${uri}/.well-known/openid-configuration`;
+    const issuerResponse = await this.authAxiosInstance.get(wellKnownUri, {
+      responseType: 'json',
+      headers: { Accept: 'application/json' },
+    });
+    return this.validateIssuer(issuerResponse.data);
+  }
+
+  public async requestAccessToken() {
```
Comment (3fd) [inline packages/api/lib/http.service.ts:119]:
```suggestion
  private async requestAccessToken(): Promise<string> {
```

File: packages/api/lib/http.service.ts [hunk 5]
```diff
@@ -87,17 +89,89 @@ export class HttpClient {
...
+    const issuer = await this.discoverIssuer(`${this.authBaseURL}/realms/${this.realm}`);
+
+    const timestamp = Date.now() / 1000;
+    const audience = [...new Set([issuer.issuer, issuer.token_endpoint].filter(Boolean))];
+
+    const encode = (input) => Buffer.from(input, 'utf8').toString('base64url');
+    const assertionPayload = {
+      iat: timestamp,
+      exp: timestamp + 60,
+      jti: encode(randomBytes(32)),
```
Comment (3fd) [inline packages/api/lib/http.service.ts:129]: can this not just be a randomUUID ?

File: packages/api/lib/token-set.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  public expiresAt: number;
+  public access_token: string;
+
+  constructor(values) {
+    Object.assign(this, values);
+  }
+
+  set expires_in(value) {
+    this.expiresAt = Date.now() + Number(value);
+  }
```
Comment (3fd) [inline packages/api/lib/token-set.ts:11]: parameter types?

File: packages/api/lib/token-set.ts (hunk 2 above)
Comment (timoklingenhoefer) [reply packages/api/lib/token-set.ts]: copied from the lib^^

**Review by 3fd — APPROVED** [review]

**Pull request merged**